src/Mantasflowers.WebApi/Controllers/UserController.cs

```diff
@@ -0,0 +1,226 @@
+                    new StringContent(content, Encoding.UTF8, "application/json")
+                    )
+                    .ConfigureAwait(false);
+
+                responseData = await response.Content.ReadAsStringAsync()
+                    .ConfigureAwait(false);
+                response.EnsureSuccessStatusCode();
+
+                string tokenId = JObject.Parse(responseData).SelectToken("idToken").ToString();
+
+                return Ok(tokenId);
+            }
+            catch (Exception ex)
+            {
+                if (ex is HttpRequestException && !string.IsNullOrEmpty(responseData))
+                {
+                    var errorData = new { error = new { code = 0, message = "errorid" } };
+                    errorData = JsonConvert.DeserializeAnonymousType(responseData, errorData);
+
+                    return Unauthorized(errorData?.error?.message ?? ResponseMsg.NotFound);
+                }
+
+                return BadRequest(ResponseMsg.NotFound);
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="role"></param>
+        /// <returns></returns>
+        [Authorize(Roles = "admin")]
+        [HttpGet("custom-role")]
+        public IActionResult SetCustomUserRole(string uid, string role)
+        {
+            var claims = new Dictionary<string, object>()
+            {
+                { ClaimTypes.Role, role },
+            };
+
+            return HandleException(async () =>
+            {
+                await _fbContext.SetCustomUserClaimsAsync(uid, claims);
+                return Ok(ResponseMsg.CustomClaimSet);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="email"></param>
+        /// <param name="password"></param>
+        /// <returns></returns>
+        [HttpGet("create-user")]
+        public IActionResult CreateUser(string email, string password)
+        {
+            return HandleException(async () =>
+            {
+                var user = await _fbContext.CreateUserAsync(email, password);
+                return Ok(user);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="uid">For testing purposes: G46RSuLIvbUjaWJyokcCMDSCeVj1</param>
+        /// <returns></returns>
+        [Authorize(Roles = "admin")]
+        [HttpGet("get-user/by-uid")]
+        public IActionResult GetUserByUid(string uid)
+        {
+            return HandleException(async () =>
+            {
+                var user = await _fbContext.GetUserByUidAsync(uid);
+                return Ok(user);
+            });
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="email">For testing purposes: johnny@john.jo</param>
+        /// <returns></returns>
+        [Authorize(Roles = "admin")]
+        [HttpGet("get-user/by-email")]
```

Comment:
If mail has to be unique in firebase, then maybe we can just have "get-user/{email}"?

Reply:
There's also "get-user/by-uid", too. Have to differentiate these endpoints somehow. What uniqueness has to do with it?

Reply:
I was initially thinking that you could differentiate them by having different parameters in route (e.g. string and guid). Buy if uid can't be a guid, then it makes sense